### DISCUSSION TITLE
pkg install quit without any message, but pkg list-installed works

### DISCUSSION THREAD

**Discussion by l1t1** [termux/termux-packages#16189] [Q&A] [answered]
apt install works
![Screenshot_2023-04-15-08-58-37-282_com termux](https://user-images.githubusercontent.com/23301704/232175419-77800c26-2e40-44d3-9b70-97dbf91ef0d0.png)

**Comment by TomJo2000** [comment]
Could you send a copy of your `pkg` script?
You can find the location using `command -v pkg`
Something must be seriously wrong with your version of `pkg`.
(Alternatively, trying updating/reinstalling `termux-tools` via apt, that should install a known good version of `pkg`)

**Reply by sylirre** [reply]
This is a real bug of `pkg` which occurs when sources.list is empty. It happens because `grep` returns error code but we have `set -e` active, which means script will be terminated on the first error returned by command.

**Reply by sylirre** [reply]
Going to be fixed in `termux-tools` update which should be available soon.

**Comment by l1t1** [comment]
I attached the pkg script to pkg.txt
[pkg.txt](https://github.com/termux/termux-packages/files/11238008/pkg.txt)
![Screenshot_2023-04-15-09-23-47-507_com termux](https://user-images.githubusercontent.com/23301704/232176727-4618a8bf-406c-47a1-8807-4a373f04afeb.png)
pkg search and pkg update and pkg upgrade also failed
![Screenshot_2023-04-15-09-11-52-649_com termux](https://user-images.githubusercontent.com/23301704/232176732-16816e66-69d2-4232-9e28-ab596257edb6.png)
apt install termux-tools didnt update the tool.
![Screenshot_2023-04-15-09-24-51-319_com termux](https://user-images.githubusercontent.com/23301704/232176770-65c342a5-b01c-4461-a14e-3426ff9d79f2.png)

**Reply by licy183** [reply, marked as answer]
You can forcedly reinstall it using `apt install --reinstall termux-tools`. Could you please also provide the output of command `bash -x $(command -v pkg) update`?

**Comment by l1t1** [comment]
bash on the bad phone
[bash.txt](https://github.com/termux/termux-packages/files/11238970/bash.txt)

bash on a good phone
[bash2.txt](https://github.com/termux/termux-packages/files/11238972/bash2.txt)

**Comment by l1t1** [comment]
/data/data/com.termux/files/usr/etc/apt/sources.list of the bad phone is empty, maybe destoried by power down
I add to it with following lines, it works
```
# The termux repository mirror from TUNA:
deb https://mirrors.tuna.tsinghua.edu.cn/termux/apt/termux-main stable main
```
